website/blog/2024-06-21-AgentEval/index.mdx

```diff
@@ -0,0 +1,182 @@
+        "successful_response": response_successful, 
+        "failed_response": response_failed, 
+    } 
+)
+
+criteria = generate_criteria(task=task, llm_config={"config_list": config_list})
+``` 
+
+Note: Only one sample execution chain (success/failure) is required for the task object but AgentEval will perform better with an example for each case. 
+
+ 
+Example Output: 
+``` 
+[
+    {
+        "name": "Accuracy",
+        "description": "The solution must be correct and adhere strictly to mathematical principles and techniques appropriate for the problem.",
+        "accepted_values": ["Correct", "Minor errors", "Major errors", "Incorrect"]
+    },
+    {
+        "name": "Conciseness",
+        "description": "The explanation and method provided should be direct and to the point, avoiding unnecessary steps or complexity.",
+        "accepted_values": ["Very concise", "Concise", "Somewhat verbose", "Verbose"]
+    },
+    {
+        "name": "Relevance",
+        "description": "The content of the response must be relevant to the question posed and should address the specific problem requirements.",
+        "accepted_values": ["Highly relevant", "Relevant", "Somewhat relevant", "Not relevant"]
+    }
+]
+``` 
+
+ 
+
+**Criteria Quantification:** 
+
+During the quantification stage, AgentEval will use the generated criteria (or user defined criteria) to assess a given execution chain to determine how well the application performed.
+
+``` 
+def quantify_criteria( 
+    llm_config: Optional[Union[Dict, Literal[False]]], 
+    criteria: List[Criterion], 
+    task: Task, 
+    test_case: str, 
+    ground_truth: str, 
+) 
+``` 
+
+Parameters: 
+* llm_config (dict or bool): llm inference configuration. 
+* criteria ([Criterion](https://github.com/microsoft/autogen/tree/main/autogen/agentchat/contrib/agent_eval/criterion.py)): A list of criteria for evaluating the utility of a given task. This can either be generated by the `generate_criteria` function or manually created.
+* task ([Task](https://github.com/microsoft/autogen/tree/main/autogen/agentchat/contrib/agent_eval/task.py)): The task to evaluate. It should match the one used during the `generate_criteria` step.
+* test_case (str): The execution chain to assess. 
+* ground_truth (str): The ground truth for the test case. 
+ 
+Example Code: 
+``` 
+quantifier_output = quantify_criteria( 
+    llm_config={"config_list": config_list}, 
+    criteria=criteria, 
+    task=task, 
+    test_case=test_case, 
+    ground_truth="true", 
+) 
+``` 
+
+The output will be a json object consisting of the ground truth and a dictionary mapping each criteria to it's score.  
+
+``` 
+{ 
+  "actual_success": true,
+  "estimated_performance": {
+      "Accuracy": "Correct",
+      "Conciseness": "Concise",
+      "Relevance": "Highly relevant"
+    } 
+} 
+```
+
+## What is next?
+* Enbabling AgentEval in AutoGen Studio for a nocode solution.
+* Fully implementing VerifierAgent in the AgentEval framework.
```

Comment:
I think we should also include a bullet point with limitations. When to use AgentEval, and when not to use it.

Reply:
In theory, AgentEval should be applicable to a wide range of applications and domains. I think the only time it wouldn't be very useful is if the user task constantly changes (though even that might still benefit with some prompt changes).

Reply:
We could not really think of many limitations. We mostly want developers to tell us how it works for their applications and what might be missing.

As for limitations identified so far from the research perspective:
- Only experimented with GPT-4.
- Tried it on limited datasets so far. It seems that human help is needed after the Critic step to verify the results, but this verification needs to be done by a domain expert.

@BeibinLi  do you think it would be beneficial to add it to the post?

Reply:
> I think we should also include a bullet point with limitations. When to use AgentEval, and when not to use it.

We could not really think of many limitations. We mostly want developers to tell us how it works for their applications and what might be missing.

As for limitations identified so far from the research perspective:
- Only experimented with GPT-4.
- Tried it on limited datasets so far. It seems that human help is needed after the Critic step to verify the results, but this verification needs to be done by a domain expert.

@BeibinLi  do you think it would be beneficial to add it to the post?

Reply:
@julianakiseleva  @jluey1  Thanks for the reply. Yes, it makes sense. Let's skip the limitation here.